tdm/docs/architecture.md

```diff
@@ -0,0 +1,222 @@
+| TEE I/O Device Interface               | TDI    | The unit of assignment for a trusted I/O capable device. For example, a TDI can be a Virtual Function (VF) or a Physical Function (PF).
+| TEE VM                                 | TVM    | A VM instantiation of an confidential workload.
+| Trusted Device Manager                 | TDM    | A confidential guest TEE-IO device manager, responsible for verifying, attesting and accepting CoVE-IO devices into a TVM TCB. This is a TVM guest software stack component.
+
+
+## Goals
+
+In order to achieve high performance Input/Output (I/O) operations, confidential
+VMs must extend their trust boundary to include a composition of directly
+assigned, TEE-IO-capable devices/functions. Without this ability, confidential
+VMs have to resort to using para-virtualized I/O using non-confidential memory
+regions, which has performance impacts due to memory copies and negates the use
+of directly assigned devices.
+
+Silicon vendors and device manufacturers provide TEE-IO compliant hardware
+implementations. Intel TDX Connect or AMD SEV-TIO, for example, rely on device
+manufacturers to support the TDISP, SPDM and PCIe IDE protocols to give
+confidential guests access to trusted MMIO and secure DMA between them and
+devices which trustworthiness is verifiable.
+
+The TEE-IO security model is built on top of this hardware support but also on
+the ability for confidential guests to ultimately decide if they accept or
+reject TEE-IO compliant devices into their TCB. The TDM, as a confidential guest
+component, is the TCB acceptance decision maker and is responsible for notifying
+the underlying Trusted Security Manager (TSM) about its decision. By accepting a
+TDI into the guest TCB, the TDM explicitly requests the TSM to move the device
+into an operational state. The TDM builds the acceptance decision after having
+verified and authenticated the TDI. It must also attest of the assigned device
+trustworthiness, through a local or remote attestation process.
+
+## Scope
+
+## Architectural Overview
+
+High-level view:
+
+```
+
+
+                                   ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+                                   ┃                            Confidential Guest ┃
+                                   ┃           ╔═══════════════════════╗           ┃
+                                   ┃           ║Trusted Device Manager ║           ┃
+                                   ┃           ║                       ║           ┃
+                                   ┃           ║┌─────────────────────┐║           ┃
+                                   ┃           ║│    Relying Party    │║           ┃
+                                   ┃      ┌───▷║│(Attestation Service)│║           ┃
+                                   ┃      │    ║└─────────────────────┘║           ┃
```

Comment:
It's a bit confusing to me to have the Relying Party / AS drawn inside the guest here. Should this be the AA, which is a trusted proxy to those services? I assume the idea is that the TDM can get trusted configuration or secrets from the client?

Also, should we differentiate between client provided trusted information and other artifacts that might come from the manufacturer (with signatures)?

Reply:
@sameo , is the Relying Party block in the diagram specific for "local attestation" scheme, where the Trusted Device Manager performs the TEE-IO device Attestation REPORT verification?

Reply:
Sorry the diagram was specific to local attestation. In my mind, the device attestation can either be local or remote and both paths should be handled through the attestation module. For local attestation, this module would expect to be provided (statically or dynamically) with attestation policies and reference values. For remote attestation, the module would use the AA API and use it as an attestation proxy. Note that this creates a dependency between the TDM and the AA (i.e. the TDM could not be used outside of the confidential-containers context without pulling the AA dependency), which is fine with me.